src/Raven.Server/Documents/TransactionOperationsMerger.cs

```diff
@@ -848,5 +1125,283 @@ public void Dispose()
+                {
+                    ReadStartRecordingDetails(readersItr, context, peepingTomStream);
+                    while (readersItr.MoveNext())
+                    {
+                        using (readersItr.Current)
+                        {
+                            if (readersItr.Current.TryGet(nameof(RecordingDetails.Type), out string strType) == false)
+                            {
+                                throw new ReplayTransactionsException($"Can't read {nameof(RecordingDetails.Type)} of replay detail", peepingTomStream);
+                            }
+
+                            if (Enum.TryParse<TxInstruction>(strType, true, out var type))
+                            {
+                                switch (type)
+                                {
+                                    case TxInstruction.BeginTx:
+                                        txDisposable = _parent.DocumentsStorage.ContextPool.AllocateOperationContext(out txCtx);
+                                        txCtx.OpenWriteTransaction();
+                                        break;
+                                    case TxInstruction.Commit:
+                                        txCtx.Transaction.Commit();
+                                        break;
+                                    case TxInstruction.DisposeTx:
+                                        txDisposable.Dispose();
+                                        break;
+                                    case TxInstruction.BeginAsyncCommitAndStartNewTransaction:
+                                        previousTx = txCtx.Transaction;
+                                        txCtx.Transaction = txCtx.Transaction.BeginAsyncCommitAndStartNewTransaction();
+                                        txDisposable = txCtx.Transaction;
+                                        break;
+                                    case TxInstruction.EndAsyncCommit:
+                                        previousTx.EndAsyncCommit();
+                                        break;
+                                    case TxInstruction.DisposePrevTx:
+                                        previousTx.Dispose();
+                                        break;
+                                }
+                                continue;
+                            }
+
+                            try
+                            {
+                                var cmd = DeserializeCommand(strType, readersItr.Current, context, peepingTomStream);
+                                cmd.ExecuteDirectly(txCtx);
+                                commandsProgress++;
+                                UpdateGlobalReplicationInfoBeforeCommit(txCtx);
+                            }
+                            catch (Exception)
+                            {
+                                //TODO To accept exceptions that was thrown while recording
```

Comment:
Reminder on this

Reply:
[RavenDB-11751](http://issues.hibernatingrhinos.com/issue/RavenDB-11751)